### PR TITLE
release v0.10.1 with critical section bug

### PR DESCRIPTION
**Pull request by almindor**
Release the critical section bugfix from #121

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+## [v0.10.1] - 2023-01-18
```
Comment (dkhayes117) [inline CHANGELOG.md:10]: I don't think this has the comparison link.

**Review by dkhayes117 — COMMENTED** [review]
I think the changelog needs the comparison link for v0.10.1

File: CHANGELOG.md (hunk 1 above)
Comment (almindor) [reply CHANGELOG.md:10]: fixed

**Review by dkhayes117 — APPROVED** [review]
Awesome, Thanks!

**Pull request merged**